ERCS/erc-7555.md

```diff
@@ -0,0 +1,211 @@
+
+### Definitions
+- **Smart account** - An ERC-4337 compliant smart contract account that has a modular architecture.
+- **Domain** - A string of text acting as an identification to a server or wesbite (eg: `ethereum.org` or `ABCDE12345.com.example.app`).
+- **EOA** - Accounts that are controlled by a single private key.
+- **Provider** - A third party service provider that is able to authenticate a user and produce a keypair for the user.
+
+### Redirects
+An application looking to authenticate a user must navigate the user to a given provider's URI based on the `URI Request Syntax`. The application must implement a valid redirect URI for the callback in order to receive a valid response.
+
+#### Available Routes
+- `/auth/`: The route used to authenticate a user, and request credentials.
+- `/sendTransaction/`: The route used to send a transaction payload for a user to sign. This is more of a convenient method to allow applications to do both authentication, and plugin registration within a single redirect, instead of requiring the user to perform two redirects.
+
+### Schema
+The `smart_account_address` should be returned in the CAIP-10 format.
+
+#### Auth Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+```
+##### Response Schema
+```=
+ parameters:
+  - in: query
+      name: smart_account_address
+      schema:
+        type: string
+      description: The on-chain address for a given smart account, formatted using CAIP-10
+```
+
+##### Request Syntax
+```=
+https://<PROVIDER_URI>/auth/?
+    redirect_uri=<YOUR_REDIRECT_URI>
+    &chain_id=<CHAIN_ID>
+```
+##### Response Syntax
+```=
+https://<YOUR_REDIRECT_URI>/auth/?
+    smart_account_address=<SMART_ACCOUNT_ADDRESS>
+```
+
+#### sendTransaction Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+   - in: query
+      name: transaction
+      schema:
+        type: string
+      description: The RLP encoded transaction that needs to be signed
+```
+##### Response Schema
+```=
+ parameters:
+  - in: query
+      name: smart_account_address
+      schema:
+        type: string
+      description: The on-chain address for a given smart account, formatted using CAIP-10
+  - in: query
+      name: tx_hash
+      schema:
+        type: string
+      description: The hash of the transaction
```

Comment:
for a 4337 account, is this the `userOpHash` or the userOp bundle's`txHash` or something else?

Reply:
and in the case of a multisig or something that wasn't expected to be completed immediately, how shoudl the provider respond here?

Reply:
hmmm this is a bit of a gap, do you have a suggestion here?

Reply:
can think about it but nothing obvious jumps to mind unf. Def think the latter should be handled, potentially would just say have two optional return params (`tx_hash`, `user_op_hash`) and if neither is returned you can assume the txn was successfully received but not completed [for any reason]. Which I guess begs a follow up question: how do you expect errors to be handled?